code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -346,6 +346,9 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 		var/turf/T = get_turf(location)
 		if(!T || T.density || !istype(location))
 			return
+		if(!location.material.isMutable()) //this is a little hacky, but basically ensure it's mutable and then do the trigger
+			location.material = location.material.getMutable()
+			return location.material.triggerTemp(location, 0)
```

Comment:
won't this just fail if you pass it 0 temp?

Reply:
for this material the same proc is used for temp and exp triggers, and in both the second argument is ignored. It is very hacky, and this continues that tradition.